test/replication-luatest/gh_6033_box_promote_demote_test.lua

```diff
@@ -0,0 +1,325 @@
+-- Test that box_promote will return box.error.UNSUPPORTED
+-- when trying to promote voter
+g.test_voter_promote = function(g)
+    local ok, err = g.server_1:exec(function()
+        return pcall(box.ctl.promote)
+    end)
+    luatest.assert(
+        not ok and err.code == box.error.UNSUPPORTED,
+        'error while promoting voter')
+end
+
+-- Test interfering promotion for election_mode = 'off'
+-- while in WAL delay
+g.test_wal_interfering_promote = function(g)
+    local wal_write_count = g.server_1:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY', true)
+        return box.error.injection.get('ERRINJ_WAL_WRITE_COUNT')
+    end)
+
+    g.server_2:exec(function()
+        box.ctl.promote()
+    end)
+    g.server_2:wait_synchro_queue_owner()
+    g.server_1:wait_wal_write_count(wal_write_count + 1)
+
+    local ok, err = g.server_1:exec(function()
+        local f = require('fiber').new(box.ctl.promote); f:set_joinable(true)
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+        return f:join()
+    end)
+    luatest.assert(
+        not ok and err.code == box.error.INTERFERING_PROMOTE,
+        'interfering promote not handled')
+end
+
+-- Test interfering promotion for election_mode = 'off'
+-- while in txn_limbo delay
+g.test_limbo_empty_interfering_promote = function(g)
+    local wal_write_count, f = g.server_1:exec(function()
+        box.error.injection.set('ERRINJ_TXN_LIMBO_EMPTY_DELAY', true)
+        local f = require('fiber').new(box.ctl.promote); f:set_joinable(true)
+        return box.error.injection.get('ERRINJ_WAL_WRITE_COUNT'), f:id()
+    end)
+
+    g.server_2:exec(function()
+        box.ctl.promote()
+    end)
+    g.server_2:wait_synchro_queue_owner()
+    g.server_1:wait_wal_write_count(wal_write_count + 1)
+
+    local ok, err = g.server_1:exec(function(f)
+        box.error.injection.set('ERRINJ_TXN_LIMBO_EMPTY_DELAY', false)
+        return require('fiber').find(f):join()
+    end, {f})
+    luatest.assert(
+        not ok and err.code == box.error.INTERFERING_PROMOTE,
+        'interfering promote not handled')
+end
+
+-- Test failing box_wait_limbo_acked in box_promote for election_mode = 'off'
+g.test_fail_limbo_acked_promote = function(g)
+    g.server_2:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY', true)
+    end)
+
+    g.server_3:box_config({
+        replication_synchro_quorum = 3,
+        replication_timeout = 0.5,
+        replication = {
+            helpers.instance_uri('server_', 1),
+        },
+    })
+
+    local wal_write_count = g.server_1:exec(function(r)
+        box.cfg{replication = {r}}
+        return box.error.injection.get('ERRINJ_WAL_WRITE_COUNT')
+    end, {helpers.instance_uri('server_', 3)})
+
+
+    g.server_3:exec(function()
+        local s = box.schema.create_space('test', {is_sync = true})
+        s:create_index('pk')
+        require('fiber').create(s.replace, s, {1})
+    end)
+    g.server_3:wait_synchro_queue_owner()
```

Comment:
Why do you need this line? You already did `g.server_3:wait_synchro_queue_owner()` in `before_each` call and there was no a reason for `server_3` to stop being the owner since the test case started.